pandas/types/missing.py

```diff
@@ -391,3 +394,36 @@ def na_value_for_dtype(dtype):
     elif is_bool_dtype(dtype):
         return False
     return np.nan
+
+
+def valid_fill_value(value, dtype, is_period=False):
+    """
+    Makes sure the fill value is appropriate for the given dtype.
+
+    Parameters
+    ----------
+    value : scalar
+    dtype: string / dtype
+    is_period : bool
```

Comment:
take this parameter out, you can skip period validation for now if its not working